src/pilot/assign_role.py

```diff
@@ -745,7 +745,7 @@ def configure_raid(ironic_client, node_uuid, super_role, os_volume_size_gb,
     when RAID configuration failed and return False. Further testing
     should uncover interesting error conditions.'''
 
-    if get_raid_controller_ids(drac_client) is None:
+    if not get_raid_controller_ids(drac_client):
```

Comment:
so this change was necessary because `get_raid_controller_ids` returned empty list `[]` and not `None`?

Reply:
@ajya, yes, that is why. The situation was encountered while testing against a systems that has no RAID controller.

When the current code finds that condition, instead of immediately terminating the configuration of RAID processing, it proceeds along until it finds the condition again. A number of confusing log messages are emitted along the way.